axiom/annotations.go

```diff
@@ -0,0 +1,172 @@
+	Description string `json:"description"`
+	// URL is the URL of the Annotation.
+	URL string `json:"url"`
+	// Type is the type of the Annotation. Can only contain lowercase
+	// alphanumerical characters and dashes.
+	Type string `json:"type"`
+}
+
+// AnnotationCreateRequest is the request payload for creating a new Annotation.
+type AnnotationCreateRequest struct {
+	// Required:
+	// Datasets is a list of dataset IDs that the Annotation is attached to.
+	Datasets []string `json:"datasets"`
+	// Type is the type of the Annotation. Can only contain lowercase
+	// alphanumerical characters and dashes.
+	Type string `json:"type"`
+	// Optional:
+	// Time is the start time of the Annotation.
+	Time time.Time `json:"time,omitempty"`
+	// EndTime is the end time of the Annotation.
+	EndTime time.Time `json:"endTime,omitempty"`
+	// Title is the title of the Annotation.
+	Title string `json:"title,omitempty"`
+	// Description is the description of the Annotation.
+	Description string `json:"description,omitempty"`
+	// URL is the URL of the Annotation.
+	URL string `json:"url,omitempty"`
+}
+
+// AnnotationUpdateRequest is the request payload for updating an Annotation.
+type AnnotationUpdateRequest struct {
+	// Datasets is a list of dataset IDs that the Annotation is attached to.
+	Datasets []string `json:"datasets,omitempty"`
+	// Time is the start time of the Annotation.
+	Time time.Time `json:"time,omitempty"`
+	// EndTime is the end time of the Annotation.
+	EndTime time.Time `json:"endTime,omitempty"`
+	// Title is the title of the Annotation.
+	Title string `json:"title,omitempty"`
+	// Description is the description of the Annotation.
+	Description string `json:"description,omitempty"`
+	// URL is the URL of the Annotation.
+	URL string `json:"url,omitempty"`
+	// Type is the type of the Annotation. Can only contain lowercase
+	// alphanumerical characters and dashes.
+	Type string `json:"type,omitempty"`
+}
+
+// AnnotationsService handles communication with the dataset related operations
+// of the Axiom API.
+//
+// Axiom API Reference: /v2/annotations
+type AnnotationsService service
+
+// Create a new annotation.
+func (a *AnnotationsService) Create(ctx context.Context, annotation *AnnotationCreateRequest) (*Annotation, error) {
+	ctx, span := a.client.trace(ctx, "Annotations.Create")
+	defer span.End()
+
+	var res Annotation
+	if err := a.client.Call(ctx, http.MethodPost, a.basePath, annotation, &res); err != nil {
+		return nil, spanError(span, err)
+	}
+
+	return &res, nil
+}
+
+// AnnotationsFilter filters annotations on the [`List`] method.
+type AnnotationsFilter struct {
+	Datasets []string   `url:"datasets"`
+	Start    *time.Time `url:"start"`
+	End      *time.Time `url:"start"`
+}
+
+// List annotations.
+func (a *AnnotationsService) List(ctx context.Context, filter *AnnotationsFilter) ([]*Annotation, error) {
+	ctx, span := a.client.trace(ctx, "Annotations.List", trace.WithAttributes())
```

Comment:
Please add attributes to `trace.WithAttributes()` or drop it. Ideally, add them conditionally if `filter != nil`.

Reply:
Fixed in 94f6afd